### PR TITLE
Button to select month + input category for expense entry

### PR DESCRIPTION
**Pull request by ChryslineLim**
- Created a button to select which month's entries to be displayed (displays 3 months prior to current month) 
- Included category into the expenses entries (only others option displayed; code commented out can only be tested once the budget backend is done)

### CODE REVIEW

File: src/screens/EditExpensesScreen.js [hunk 1]
```diff
@@ -1,22 +1,25 @@
-import { SafeAreaView, Text, StyleSheet, TextInput, Pressable, Button } from "react-native";
+import { SafeAreaView, Text, StyleSheet, TextInput, Pressable } from "react-native";
 import React, { useState } from 'react';
 import { db, auth } from "../firebase";
 import { collection, addDoc } from "firebase/firestore";
 import { DatePicker } from "../Components/DatePicker";
+import { CategoryPicker } from "../Components/CategoryPicker";
```
Comment (Jiacheng-y) [inline src/screens/EditExpensesScreen.js:6]: perhaps these import statement can be combined :

`import { DatePicker, CategoryPicker } from "../Components"; `

File: src/functions/monthName.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            break;
+        case 11:
+            name = "November";
+            break;
+        case 12:
+            name = "December";
+            break;
+    }
+    return name;
+}
```
Comment (Jiacheng-y) [inline src/functions/monthName.js:42]: I really appreciate the separation of the function monthName from components

File: src/screens/ExpensesScreen.js [hunk 3]
```diff
@@ -3,17 +3,19 @@ import React, { useState, useEffect } from 'react';
 import { db, auth } from "../firebase";
 import { query, collection, doc, onSnapshot, deleteDoc } from "firebase/firestore";
 import { ExpenseEntry } from '../Components/ExpenseEntry';
+import { MonthPicker } from "../Components/MonthPicker";
```
Comment (Jiacheng-y) [inline src/screens/ExpensesScreen.js:6]: perhaps these can be combined into 

`import { MonthPicker, ExpenseEntry } from "../Components";`

File: src/Components/MonthPicker.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        default:
+            month3 = currentMonth - 3;
+    }
+
+    const [open, setOpen] = useState(false);
+    const [items, setItems] = useState([
+        {label: 'Month', value: 'Month'},
+        {label: `${monthName(month3)}`, value: `${month3}`, parent: 'Month'},
+        {label: `${monthName(month3 + 1 % 12)}`, value: `${month3 + 1 % 12}`, parent: 'Month'},
+        {label: `${monthName(month3 + 2 % 12)}`, value: `${month3 + 2 % 12}`, parent: 'Month'},
```
Comment (Jiacheng-y) [inline src/Components/MonthPicker.js:28]: I might be wrong but `month3 + 1 % 12` seems to be the same as `month3 + 1`

File: src/Components/MonthPicker.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        {label: `${monthName(currentMonth)}`, value: `${currentMonth}`, parent: 'Month'},
+
+        {label: 'Year', value: 'Year'},
```
Comment (Jiacheng-y) [inline src/Components/MonthPicker.js:31]: Just to check that I understood this correctly : we allow the user to choose any of the previous 3 months to view their expenses

**Review by Jiacheng-y — APPROVED** [review]
The expense screens, added functionalities and components look great! :)

File: src/Components/MonthPicker.js (hunk 4 above)
Comment (ChryslineLim) [reply src/Components/MonthPicker.js:28]: It's to wrap around the months! For example if the month3 is December, then I want the next month to be January and not an unknown month 13

File: src/Components/MonthPicker.js (hunk 5 above)
Comment (ChryslineLim) [reply src/Components/MonthPicker.js:31]: That's right! I included the previous year as well for the case of any of the previous 3 months being in the preceding year

File: src/screens/EditExpensesScreen.js (hunk 1 above)
Comment (ChryslineLim) [reply src/screens/EditExpensesScreen.js:6]: I've made the changes!

File: src/screens/ExpensesScreen.js (hunk 3 above)
Comment (ChryslineLim) [reply src/screens/ExpensesScreen.js:6]: I've made the changes!

**Pull request merged**